tensorflow/python/ops/array_ops.py

```diff
@@ -1917,7 +1917,6 @@ def one_hot(indices, depth, on_value=1, off_value=0,
             axis, dtype], name, "one_hot") as name:
     on_value = ops.convert_to_tensor(on_value, dtype=dtype, name="on_value")
```

Comment:
No. They should with with tensors.  Sounds like a bug in cast.  Let's fix
that either by adding support for cast integers to complex or remove
dtype=complex from the test.
On May 10, 2016 12:47 PM, "Sam Abrahams" <notifications@github.com> wrote:

> In tensorflow/python/ops/array_ops.py
> <https://github.com/tensorflow/tensorflow/pull/2187#discussion_r62739194>:
>
> > @@ -1917,7 +1917,6 @@ def one_hot(indices, depth, on_value=1, off_value=0,
> >              axis, dtype], name, "one_hot") as name:
> >      on_value = ops.convert_to_tensor(on_value, dtype=dtype, name="on_value")
>
> @ebrevdo <https://github.com/ebrevdo> When these are switched to cast, we
> end up with an UnimplementedError when trying to cast standard Python
> numbers to complex64, while convert_to_tensor has code coverage for this.
> Might we stick with convert_to_tensor for now and switch after the
> appropriate cast implementations are covered? From what I understand, the
> main downside of convert_to_tensor is that it may not work properly when
> on_value and off_value are passed in as Tensor objects, but I imagine
> most people just passing in normal Python types.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/tensorflow/tensorflow/pull/2187/files/d5973cced2fb296b81dd65226211d4c109575841#r62739194>
>